windows-agent/internal/proservices/landscape/distroinstall/distroinstall.go

```diff
@@ -16,6 +16,15 @@ import (
 	"github.com/ubuntu/gowsl"
 )
 
+// CommandNotFoundError is the error reported when a command is not found.
+type CommandNotFoundError struct {
+	errMsg string
+}
+
+func (e *CommandNotFoundError) Error() string {
```

Comment:
Could this be this instead?

```suggestion
func (e CommandNotFoundError) Error() string {
```

Reply:
While it could for the purposes we use here, it's a best practice to make the  `Error()` method with a pointer-receiver. That has to do with how type-assertions work. If we'd make that a value-receiver instead, as you're suggesting, both expressions would be compilable: `if var err error = SomeFun(); err.(CommandNotFoundError)` and `if var err error = SomeFun(); err.(*CommandNotFoundError)` but one of them would be broken, depending on whether `func SomeFunc() error` returned a pointer or a value (both implementations would be valid from the compiler's perspective). It's a quite obscure problem, to be honest. So sticking to the convention allows the error handler to not care about the possibility of the `err` containing a value or a pointer. It's a pointer (or nil if no error) and that's it.

By making the `Error()` method with a pointer-receiver we constrain the type system such that the `CommandNotFoundError` type doesn't satisfy the `error` interface but the pointer to it (`*CommandNotFoundError`) does. Thus, checking for the value type in a type assertion makes no sense.

This link demonstrates how that's possible. 

https://go.dev/play/p/FYiyBUdlSvZ

Reply:
Ah that makes sense, ignore my suggestion then :)